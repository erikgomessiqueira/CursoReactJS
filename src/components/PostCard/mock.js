export const postCardPropsMock = {
  title: 'title01',
  body: 'body01',
  id: 1,
  cover: 'img/img.png',
};
