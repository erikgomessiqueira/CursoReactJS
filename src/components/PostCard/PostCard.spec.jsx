import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { postCardPropsMock } from './mock';

const props = postCardPropsMock;

describe('<PostCard/>', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard {...props} />);

    const img = screen.getByAltText(/title01/i);
    expect(img).toHaveAttribute('src', 'img/img.png');

    const heading = screen.getByRole('heading', { name: 'title01 1' });
    expect(heading).toBeInTheDocument();

    const body = screen.getByText('body01');
    expect(body).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
