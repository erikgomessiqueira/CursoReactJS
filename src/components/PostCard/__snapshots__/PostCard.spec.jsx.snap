// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard/> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title01"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h1>
      title01
       
      1
    </h1>
    <p>
      body01
    </p>
  </div>
</div>
`;
